core/chains/evm/logpoller/log_poller.go

```diff
@@ -306,6 +302,9 @@ func (lp *logPoller) Filter(from, to *big.Int, bh *common.Hash) ethereum.FilterQ
 // Replay signals that the poller should resume from a new block.
 // Blocks until the replay is complete.
 // Replay can be used to ensure that filter modification has been applied for all blocks from "fromBlock" up to latest.
+// If ctx is cancelled before the replay request has been initiated, ErrReplayAbortedByClient is returned.  If the replay
+// is already in progress, the replay will continue and ErrReplayInProgress will be returned.  If the client needs a
+// guarantee that the replay is complete before proceeding, it must wait for a nil return value.
```

Comment:
What does 'wait for a nil return value' mean?

Reply:
It means that if the caller cares about the replay being complete, it should either avoid cancelling the context or keep retrying until it gets a nil return value.

Reply:
Can we clarify the language? 'wait' sounds like (a)sync/(non)blocking terminology and sent me looking for `select`s.